hydradx/model/hollar.py

```diff
@@ -30,26 +33,30 @@ def __init__(
 
         if len(pools) != len(self.asset_list):
             raise ValueError("pools must have same length as asset_list")
-        self.pools = {}
+        if len({id(pool) for pool in pools}) != len(pools):
+            raise ValueError("pools must be unique")
+        self.pools = {}  # references to designated stableswap pools
+        self._pool_states = {}  # copy of pool states used for calculations
```

Comment:
This seems to have no point - copying each pool on every update is slow, and nothing is ever done with them that would mutate their state, so you might as well just use the original objects.

Reply:
The pool states change when they are traded with. Note that self.pools holds a reference to designated stablepools, so they can be changed outside of the HSM object (and we expect they will be changed this way). We need to track the original pool state because that is what the HSM functionality will be based on.

Reply:
I have created a ticket to potentially increase the efficiency of this calculation, but I will leave it as-is for this PR.